core/src/main/java/uk/gov/gchq/koryphe/impl/function/IterableFlatten.java

```diff
@@ -54,6 +55,7 @@ public I_ITEM apply(final Iterable<I_ITEM> items) {
         if (nonNull(items) && nonNull(operator)) {
             return StreamSupport.stream(items.spliterator(), false)
                     .onClose(() -> CloseableUtil.close(items))
+                    .filter(Objects::nonNull)
```

Comment:
For any Iterable which is being flattened, this will remove all `null` Object references **before** they are passed to the `BinaryOperator`. This prevents the problem highlighted in #261, however this prevents any `BinaryOperator` from being used with `null`.

There have been some doubts about whether this is a good idea, in case a `BinaryOperator` existed that wanted to see `null` values. However, the way that `null` is handled by `KorypheBinaryOperator` (does not run `_apply`) would make it inconvenient or impossible to create a `BinaryOperator` which did anything with `null`, other than ignoring them - it impossible to have a 'Count' `BinaryOperator` which could include `null` in a count.

This change may actually improve performance, because removing all `null` values before running the reduce will result in fewer operations.